pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -827,25 +990,19 @@ installPhase:
 					ppln.deleteWorkerFromPipelinePhase(phaseInfo.Phase, installWorker)
 				} else if phaseInfo.Status == enterpriseApi.AppPkgMissingOnPodError {
 					ppln.transitionWorkerPhase(installWorker, enterpriseApi.PhaseInstall, enterpriseApi.PhasePodCopy)
-				} else if checkIfWorkerIsEligibleForRun(installWorker, phaseInfo, enterpriseApi.AppPkgInstallComplete) {
+				} else if checkIfWorkerIsEligibleForRun(installWorker, phaseInfo, enterpriseApi.AppPkgInstallComplete) &&
+					getInstallSlotForPod(podInstallTracker, installWorker) {
```

Comment:
maybe only pass `targetPodName` here since that is the only field you are using inside this function